src/components/TodoInput/TodoInput.tsx

```diff
@@ -0,0 +1,58 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { ErrorAction } from '../../types/ErrorAction';
+
+type Props = {
+  isButtonActive: boolean;
+  isInputDisabled: boolean;
+  setError: (action: ErrorAction) => void;
+  addTodo: (title: string) => Promise<void>;
+};
+
+export const TodoInput: React.FC<Props> = React.memo(({
+  isButtonActive,
+  isInputDisabled,
+  setError,
+  addTodo,
+}) => {
+  const [input, setInput] = useState('');
+
+  const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const title = input.trim();
+
+    if (!title) {
+      setError(ErrorAction.EMPTY);
+    } else {
+      addTodo(title);
```

Comment:
Don't forget to handle the Promise returned by 'addTodo' with a try-catch block here as well
```suggestion
     await addTodo(title);
```